parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetReaders.java

```diff
@@ -75,6 +78,16 @@ protected ParquetValueReader<T> createReader(
     }
   }
 
+  /**
+   * This method can be overridden to provide a custom implementation which also uses the fieldId of
+   * the Schema when creating the struct reader
+   */
+  protected ParquetValueReader<T> createStructReader(
```

Comment:
This is key for allowing InternalParquet to use the field Id to lookup what structlike class should be used for a particular field. This also is the reason for the RevAPI breaks

Reply:
A little weird that revapi complains about API change for non-public class.

Should we just modify the original method in line 91 directly? there are only two child classes `InternalReader` and `GenericParquetReaders`. Then we can avoid the fallback behavior, which is a bit odd to me.

Reply:
The complaint is that while we are source compatible it "may" be binary incompatible